### PR TITLE
Fix tests and fix posts and comments bug

### PR DESCRIPTION
**Pull request by karacasoft**
## About

To fix #171, I thought we should fix the tests first. There were a lot of weak assertions and simply wrong test cases. ~~There are also missing cases, but I'll go through them on the next iteration of fixing #171.~~ I've fixed the Post and Comments not being created issue.

~~On the way, I fixed some logic on Comments which was taking 'owner' as an argument from outside. I have changed it to use the currently logged in user. I want the frontend team to check if their comment system is working.~~

I've changed the way both Comments and Posts work now. I want frontend to be checked if API is being used for Posts and Comments. To fix any issues, you have to send authorization information to API. Either session or Token authorization is accepted. API will request authorization at some point so getting this set up will actually buy you time from future.

### Changes

- Added exception handlers for JSON data accesses.
- Removed/Merged unnecessary cases, assertions
- Added a dummy data generator module
- Added descriptive error messages

- Added post create/update/delete tests.
- Added another Post Serializer for create and update operations

### CODE REVIEW

File: api/interestr/api/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+from . import models
+
```
Comment (smddzcy) [inline api/interestr/api/test_utils.py:2]: Cool utils 👌

**Review by smddzcy — DISMISSED** [review]
:shipit: if it works, the code looks fine.

File: api/interestr/api/test_utils.py (hunk 1 above)
Comment (karacasoft) [reply api/interestr/api/test_utils.py:2]: Good point :+1:

**Comment by karacasoft** [comment]
I actually want someone to test the frontend. So will let this stay for a while.

File: api/interestr/api/serializers.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 from . import models as core_models
 from django.contrib.auth import models as auth_models
 
+from serializer_utils import get_or_none
+
```
Comment (Morgazipa) [inline api/interestr/api/serializers.py]: I suspect this got uncaught from attention when deleting "serializer_utils". Things are fine after deleting this line.

File: api/interestr/api/tests.py [hunk 3]
```diff
@@ -16,6 +16,9 @@
 
 import json
 
+from . import test_utils
+from test_utils import responseErrorMessage as responseError
```
Comment (Morgazipa) [inline api/interestr/api/tests.py]: Here,  I had to change it to "from .test_utils import ..." to make it work.

**Review by Morgazipa — COMMENTED** [review]
Very nice work. I really liked "test_utils" and helpful error messages in tests. I also tested the frontend after making the small changes mentioned above, everything works as excepted!

File: api/interestr/api/serializers.py (hunk 2 above)
Comment (karacasoft) [reply api/interestr/api/serializers.py]: Interesting, I thought I had deleted this... lol

**Comment by karacasoft** [comment]
I did the changes @Morgazipa requested. Waiting for the next review...

**Review by Morgazipa — APPROVED** [review]
Now everything seems to work fine.

**Pull request merged**